### DISCUSSION TITLE
Migrating Wazuh manager to Distributed from Standalone

### DISCUSSION THREAD

**Discussion by mskarshinski** [Security-Onion-Solutions/securityonion#7677] [Unsupported Versions] [answered]
I think this is probably a "no", but is there a way to migrate an existing Wazuh manager, along with its registrations to a newly installed distributed setup?  I found the sslmanager key pair in /nsm/wazuh/etc and updated those to match my old keys, but I suspect the agent registrations are stored in a mysql database.  I don't mind migrating sql-sql, but this is a complex product, and I don't want to end up building in future problems.  Its not a giant hardship to redeploy my agents if need be.

Thank you!

**Comment by mskarshinski** [comment]
I think I've answered this question, at least partially using a posted discussion on general Wazuh migration, looks like Wazuh is self contained in the (in the case of SO) /nsm/wazuh folder.  I'm still looking for advice from anyone who has done this, I'm sure there are caveats.

Thanks!

**Comment by dougburks** [comment, marked as answer]
Yes, the Wazuh server is in `/nsm/wazuh/`. You might be able to migrate Wazuh agents from one manager to another, but it's not something we support here. If it's not a giant hardship to redeploy the agents, then that's what we'd recommend.

**Reply by mskarshinski** [reply]
Thank you for replying.  I was able to convince my agents to speak to the new server, and for my silent installer to register new agents successfully.  It seems to be working but I like to reserve judgement on these things until some time has passed.  I copied the deployed agents list and shared secret file to the new installation, then pushed an updated ossec.conf to the endpoints through my RMM.  I think the hardship of redeploy would have been simple, but I gained a lot of knowledge by getting down in the dirt with manual configuration.  Doesn't look like I smashed anything.  yet.
